projects/step-frontend/src/lib/modules/_common/components/sidebar/sidebar.component.ts

```diff
@@ -0,0 +1,119 @@
+    public _viewStateService: ViewStateService,
+    public _location: Location
+  ) {
+    // Main Menus
+    this._viewRegistryService.registerMenuEntry('Automation', 'automation-root', 'glyphicon glyphicon-play');
+    this._viewRegistryService.registerMenuEntry('Execute', 'execute-root', 'glyphicon glyphicon-tasks');
+    this._viewRegistryService.registerMenuEntry('Status', 'status-root', 'glyphicon glyphicon-ok');
+
+    // Sub Menus Automation
+    this._viewRegistryService.registerMenuEntry(
+      'Keywords',
+      'functions',
+      'glyphicon glyphicon-record',
+      'automation-root'
+    );
+    this._viewRegistryService.registerMenuEntry('Plans', 'plans', 'glyphicon glyphicon-file', 'automation-root');
+    this._viewRegistryService.registerMenuEntry(
+      'Parameters',
+      'parameters',
+      'glyphicon glyphicon-list-alt',
+      'automation-root'
+    );
+    // Sub Menus Execute
+    this._viewRegistryService.registerMenuEntry(
+      'Executions',
+      'executions',
+      'glyphicon glyphicon-tasks',
+      'execute-root'
+    );
+    this._viewRegistryService.registerMenuEntry('Scheduler', 'scheduler', 'glyphicon glyphicon-time', 'execute-root');
+    // Sub Menus Status
+    this._viewRegistryService.registerMenuEntry('Agents', 'gridagents', 'glyphicon glyphicon-briefcase', 'status-root');
+    this._viewRegistryService.registerMenuEntry('Agent tokens', 'gridtokens', 'glyphicon glyphicon-tag', 'status-root');
+    this._viewRegistryService.registerMenuEntry(
+      'Token Groups',
+      'gridtokengroups',
+      'glyphicon glyphicon glyphicon-tags',
+      'status-root'
+    );
+    this._viewRegistryService.registerMenuEntry(
+      'Quota Manager',
+      'gridquotamanager',
+      'glyphicon glyphicon-road',
+      'status-root'
+    );
+    this.mainMenuEntriesAmount = _viewRegistryService.getCustomMainMenuEntries().length - 1;
+  }
+
+  ngAfterViewInit(): void {
+    let initialViewName = this._viewStateService.getViewName();
+    if (initialViewName) {
+      let initiallyExpandedMainMenu = this._viewRegistryService.getMainMenuKey(initialViewName!);
+      if (initiallyExpandedMainMenu) {
+        this.openMainMenu(initiallyExpandedMainMenu);
+      }
+    }
+  }
+
+  private openMainMenu(mainMenuName: string): void {
+    for (let checkbox of this.mainMenuCheckBoxes!) {
+      let name = checkbox.nativeElement.getAttribute('name');
+      if (name === mainMenuName) {
+        checkbox.nativeElement.setAttribute('checked', true);
+      }
+    }
+  }
+
+  public navigateTo(viewId: string): void {
+    switch (viewId) {
+      case 'home':
+        this._authService.gotoDefaultPage();
+        break;
+      default:
+        this._location.go('#/root/' + viewId);
+        break;
+    }
+  }
+
+  /**
+   * FIXME: communicate open/close state as output to parent component instead of selecting global dom element
+   */
+  toggleOpenClose() {
+    if (this.sideBarOpen) {
```

Comment:
This code works, and I think we can continue to use it, while the app is hybrid.
But I would prefer to reference to document in `angular way` by injecting it:
```
constructor(
    ...
    @Inject(DOCUMENT) private _doc: Document
) {
  ....
}

...

toggleOpenClose(): void {
   ...
   this._doc.querySelector('#main')...
   ...
}

...


```

Reply:
You are right. But as Gilbert mentioned I would actually prefer to change this after the migration to not use document context at all.